packages/plugin-core/src/test/suite-integ/PasteLink.test.ts

```diff
@@ -0,0 +1,153 @@
+
+    test("basic", (done) => {
+      onWSInit(async () => {
+        const notePath = path.join(vaultPath, "foo.md");
+        await VSCodeUtils.openFileInEditor(vscode.Uri.file(notePath));
+        const link = "https://stackoverflow.com/questions/tagged/jest";
+
+        try {
+          clipboardy.writeSync(link);
+        } catch (err) {
+          throw err;
+        }
+        const metaData = await new PasteLinkCommand().run();
+
+        assert.strictEqual(
+          metaData,
+          "[Newest 'jest' Questions](https://stackoverflow.com/questions/tagged/jest)"
+        );
+        done();
+      });
+
+      setupDendronWorkspace(root.name, ctx, {
+        lsp: true,
+        useCb: async (vaultDir) => {
+          vaultPath = vaultDir;
+          await NodeTestPresetsV2.createOneNoteOneSchemaPreset({ vaultDir });
+        },
+      });
+    });
+    test("basic2", (done) => {
+      onWSInit(async () => {
+        const notePath = path.join(vaultPath, "foo.md");
+        await VSCodeUtils.openFileInEditor(vscode.Uri.file(notePath));
+        const link = "https://en.wikipedia.org/wiki/Computer_programming";
+
+        try {
+          clipboardy.writeSync(link);
+        } catch (err) {
+          throw err;
+        }
+        const metaData = await new PasteLinkCommand().run();
+
+        assert.strictEqual(
+          metaData,
+          "[Computer programming - Wikipedia](https://en.wikipedia.org/wiki/Computer_programming)"
+        );
+        done();
+      });
+      setupDendronWorkspace(root.name, ctx, {
+        lsp: true,
+        useCb: async (vaultDir) => {
+          vaultPath = vaultDir;
+          await NodeTestPresetsV2.createOneNoteOneSchemaPreset({ vaultDir });
+        },
+      });
+    });
+    test("advanced", (done) => {
+      onWSInit(async () => {
+        const notePath = path.join(vaultPath, "foo.md");
+        await VSCodeUtils.openFileInEditor(vscode.Uri.file(notePath));
+        const link = "https://www.npmjs.com/package/open-graph-scraper";
+
+        try {
+          clipboardy.writeSync(link);
+        } catch (err) {
+          throw err;
+        }
+        const metaData = await new PasteLinkCommand().run();
+
+        const ogs = require("open-graph-scraper");
+        const options = { url: link };
+        ogs(options).then((data) => {
+          const { error, result, response } = data;
+          const { ogUrl, ogSiteName, ogTitle, ogDescription, ogImage } = result;
+          const actualMetaData = `[${ogTitle}](${ogUrl})`;
+          assert.strictEqual(metaData, actualMetaData);
+          done();
+        });
+      });
+      setupDendronWorkspace(root.name, ctx, {
+        lsp: true,
+        useCb: async (vaultDir) => {
+          vaultPath = vaultDir;
+          await NodeTestPresetsV2.createOneNoteOneSchemaPreset({ vaultDir });
+        },
+      });
+    });
+
+    test("error", (done) => {
+      onWSInit(async () => {
+        const notePath = path.join(vaultPath, "foo.md");
```

Comment:
since all the tests have the same logic (write link to clipboard and call `PasteLink` on contents), I would extract it all in a helper function to make it easier to tweak this code in the future as well as to cut down on number of lines